stub/src/main/java/io/grpc/stub/ServerCallStreamObserver.java

```diff
@@ -143,4 +143,21 @@ public void disableAutoRequest() {
    */
   @Override
   public abstract void setMessageCompression(boolean enable);
+
+  /**
+   * Sets a {@link Runnable} to be called when the call is successfully completed from the server's
+   * point of view: all the messages have been put on the wire and the stream has been closed.
```

Comment:
We should have language saying that only one of onSuccessHandler/onCancelHandler will be called.

We should call out that it doesn't matter whether SO.onError() or SO.onComplete() had been called when responding to the client; as long as the server's "response" (not a message, but the actions) has been communicated, that means "success."

Reply:
ah, I completely forgot about this handler getting called after `onError(...)`...   In such case the name `onSuccessHandler` is quite confusing  ;-]   (`onSuccess` called after `onError`?? ;-)  )
I've renamed the whole thing to `onFinish`: let me know if this is ok for you.

I've also updated javadoc after your suggestions.